src/main/java/org/akhq/repositories/RecordRepository.java

```diff
@@ -140,6 +140,25 @@
         }, "Offsets for " + partitions + " Timestamp " + timestamp, null);
     }
 
+    public Optional<Record> consumeSingleRecord(String clusterId, Topic topic, Options options) throws ExecutionException, InterruptedException {
+        return Debug.call(() -> {
+            Optional<Record> singleRecord = Optional.empty();
```

Comment:
maybe tune the option `max.poll.records` to 1 to avoid to poll too many message ?

Reply:
yes, missing that.